src/cli/cli_network_data.cpp

```diff
@@ -0,0 +1,343 @@
+/*
+ *  Copyright (c) 2020, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file implements a simple CLI for the CoAP service.
+ */
+
+#include "cli_network_data.hpp"
+
+#include <openthread/border_router.h>
+#include <openthread/server.h>
+
+#include "cli/cli.hpp"
+#include "common/encoding.hpp"
+
+using ot::Encoding::BigEndian::HostSwap16;
+
+namespace ot {
+namespace Cli {
+
+const struct NetworkData::Command NetworkData::sCommands[] = {
+    {"help", &NetworkData::ProcessHelp},
+#if OPENTHREAD_CONFIG_BORDER_ROUTER_ENABLE || OPENTHREAD_CONFIG_TMF_NETDATA_SERVICE_ENABLE
+    {"register", &NetworkData::ProcessRegister},
+#endif
+    {"show", &NetworkData::ProcessShow},
+    {"steeringdata", &NetworkData::ProcessSteeringData},
+};
+
+NetworkData::NetworkData(Interpreter &aInterpreter)
+    : mInterpreter(aInterpreter)
+{
+}
+
+void NetworkData::OutputPrefix(const otBorderRouterConfig &aConfig)
+{
+    mInterpreter.OutputFormat("%x:%x:%x:%x::/%d ", HostSwap16(aConfig.mPrefix.mPrefix.mFields.m16[0]),
+                              HostSwap16(aConfig.mPrefix.mPrefix.mFields.m16[1]),
+                              HostSwap16(aConfig.mPrefix.mPrefix.mFields.m16[2]),
+                              HostSwap16(aConfig.mPrefix.mPrefix.mFields.m16[3]), aConfig.mPrefix.mLength);
+
+    if (aConfig.mPreferred)
+    {
+        mInterpreter.OutputFormat("p");
+    }
```

Comment:
nit: maybe less code size? 
```suggestion
    mInterpreter.OutputFormat("%s%s%s%s%s%s%s%s%s", 
        aConfig.mPreferred ? "p" : "", 
        aConfig.mSlaac ? "a" : "", 
        aConfig.mDhcp ? "d" : "", 
        aConfig.mConfigure ? "c" : "", 
        aConfig.mDefaultRoute ? "r" : "",
        aConfig.mOnMesh ? "o" : "",
        aConfig.mStable ? "s" : "",
        aConfig.mNdDns ? "n" : "",
        aConfig.mDp ? "D" : "",
    );
```

Reply:
I tried your suggestion and it increased code size, according to size-bot. You can see size-bot's comment history to view the difference.

Reply:
:+1: 
It's difficult to tell which is of smaller code size. 
Thanks